### PR TITLE
speeding up super-droplet count product. closes #975

### LINKED ISSUE #975: optimize `cell_start` iteration in `SUperDropletCountPerGridbox` product
it uses Storage `__getitem__` (a lot!) which is meant just for debugging,
there is also a loop that likely should be @njitted

Thanks @mstach60161 for profiling and noticing it!

### PR DESCRIPTION
**Pull request by slayoo**

### CODE REVIEW

File: PySDM/products/housekeeping/super_droplet_count_per_gridbox.py [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+    def register(self, builder):
+        super().register(builder)
+
+        @numba.njit(**{**JIT_FLAGS, "fastmath": builder.formulae.fastmath})
+        def impl(cell_start, buffer_ravel):
+            n_cell = cell_start.shape[0] - 1
+            for i in numba.prange(n_cell):  # pylint: disable=not-an-iterable
+                buffer_ravel[i] = cell_start[i + 1] - cell_start[i]
+
+        self.impl = impl
```
Comment (abulenok) [inline PySDM/products/housekeeping/super_droplet_count_per_gridbox.py:24]: It seems that it should look like this:
```suggestion
        self._impl = impl
```

https://github.com/open-atmos/PySDM/blob/a1f4b54b348d5626080fa146fbb92e5be869c2ef/PySDM/products/impl/product.py#L101

File: PySDM/products/housekeeping/super_droplet_count_per_gridbox.py [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
 
-    # TODO #992: should be njitted (with prange)
     def _impl(self, **kwargs):
-        cell_start = self.particulator.attributes.cell_start
-        n_cell = cell_start.shape[0] - 1
-        for i in range(n_cell):
-            self.buffer.ravel()[i] = cell_start[i + 1] - cell_start[i]
+        self.impl(
+            cell_start=self.particulator.attributes.cell_start.to_ndarray(),
+            buffer_ravel=self.buffer.ravel(),
```
Comment (abulenok) [inline PySDM/products/housekeeping/super_droplet_count_per_gridbox.py:29]: It seems to me that `raveled_buffer` would better reflect the purpose of this variable

**Review by abulenok — CHANGES_REQUESTED** [review]

File: PySDM/products/housekeeping/super_droplet_count_per_gridbox.py (hunk 1 above)
Comment (slayoo) [reply PySDM/products/housekeeping/super_droplet_count_per_gridbox.py:24]: `_impl` and `impl` are completely different routines! Which indeed calls for a name change...

File: PySDM/products/housekeeping/super_droplet_count_per_gridbox.py (hunk 2 above)
Comment (slayoo) [reply PySDM/products/housekeeping/super_droplet_count_per_gridbox.py]: done

File: PySDM/products/housekeeping/super_droplet_count_per_gridbox.py (hunk 1 above)
Comment (slayoo) [reply PySDM/products/housekeeping/super_droplet_count_per_gridbox.py]: done

**Review by abulenok — APPROVED** [review]

**Pull request merged**